src/main/java/it/gov/pagopa/pu/workflow/wf/exportfile/wfexportfile/ExportFileWFImpl.java

```diff
@@ -0,0 +1,115 @@
+    scheduleExportFileExpirationActivity = wfConfig.buildScheduleExportFileExpirationActivityStub();
+  }
+
+  @Override
+  public void exportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    log.info("Starting export file workflow for exportFileId: {}", exportFileId);
+    boolean success = processExportFile(exportFileId, exportFileType);
+    if(success){
+      scheduleExportFileExpiration(exportFileId);
+    }
+    //TODO sendEmailActivity will be added with the task P4ADEV-2597
+    log.info("Completed export file workflow for exportFileId: {}", exportFileId);
+  }
+
+  private boolean processExportFile(Long exportFileId, ExportFileTypeEnum exportFileType) {
+    updateExportFileStatus(UpdateStatusRequest.builder()
+      .exportFileId(exportFileId)
+      .oldStatus(ExportFileStatus.REQUESTED)
+      .newStatus(ExportFileStatus.PROCESSING)
+      .build());
+    Pair<ExportFileResult,String> exportResultPair = executeExport(exportFileId, exportFileType);
+    ExportFileResult exportFileResult = exportResultPair.getLeft();
+    String errorDescription = exportResultPair.getRight();
+    updateExportFileWithProcessingResult(exportFileId, errorDescription, exportFileResult);
+    return StringUtils.isBlank(errorDescription);
+  }
+
+  private void updateExportFileWithProcessingResult(Long exportFileId, String errorDescription,
+    ExportFileResult exportFileResult) {
+    UpdateStatusRequest updateStatusRequest;
+    if(StringUtils.isBlank(errorDescription)){
+      updateStatusRequest = UpdateStatusRequest.builder()
+        .exportFileId(exportFileId)
+        .oldStatus(ExportFileStatus.PROCESSING)
+        .newStatus(ExportFileStatus.COMPLETED)
+        .filePathName(exportFileResult.getFilePath())
+        .fileName(exportFileResult.getFileName())
+        //TODO field fileSize depends on task P4ADEV-2598
+        .exportedRows(exportFileResult.getExportedRows())
+        .build();
+    }else{
+      updateStatusRequest = UpdateStatusRequest.builder()
+        .exportFileId(exportFileId)
+        .oldStatus(ExportFileStatus.PROCESSING)
+        .newStatus(ExportFileStatus.ERROR)
+        .errorDescription(errorDescription)
+        .build();
+    }
+    updateExportFileStatus(updateStatusRequest);
+  }
+
+  private void updateExportFileStatus(UpdateStatusRequest updateStatusRequest) {
+    updateExportFileStatusActivity.updateStatus(updateStatusRequest);
+  }
+
+  private Pair<ExportFileResult,String> executeExport(Long exportFileId, ExportFileTypeEnum exportFileType){
+    try {
+      return Pair.of(exportFileActivity.executeExport(
+          exportFileId, exportFileType),null);
+    }catch(Exception e){
+      return Pair.of(null,e.getMessage());
+    }
+  }
+
+  private void scheduleExportFileExpiration(Long exportFileId) {
+    scheduleExportFileExpirationActivity.scheduleExportFileExpiration(
+      exportFileId,
+      LocalDate.now().plusDays(expirationDays)
```

Comment:
we should store this date on ExportFile!
maybe we could use the updateStatus itself?